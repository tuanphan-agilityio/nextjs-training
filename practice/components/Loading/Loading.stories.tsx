import { Meta } from '@storybook/react';
import { FC } from 'react';

import Loading from '.';

export default {
  title: 'Loading',
  component: Loading,
} as Meta;

export const Default: FC = () => <Loading />;
