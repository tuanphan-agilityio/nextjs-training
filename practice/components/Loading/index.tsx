import { FC } from 'react';

const Loading: FC = () => {
  return (
    <div className='fixed top-0 right-0 bottom-0 left-0 flex items-center justify-center bg-tertiary z-50'>
      <span
        className='inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite]'
        role='status'
      />
    </div>
  );
};

export default Loading;
