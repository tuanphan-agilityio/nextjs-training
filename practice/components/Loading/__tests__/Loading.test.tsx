import { render } from '@testing-library/react';

import Loading from '..';

describe('Loading Component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });

  it('should have the correct class name', () => {
    const { container } = render(<Loading />);
    const loadingElement = container.querySelector('.bg-tertiary');
    expect(loadingElement).toBeInTheDocument();
  });

  it('should have the spinning animation', () => {
    const { container } = render(<Loading />);
    const spinnerElement = container.querySelector('.animate-spin');
    expect(spinnerElement).toBeInTheDocument();
  });
});
