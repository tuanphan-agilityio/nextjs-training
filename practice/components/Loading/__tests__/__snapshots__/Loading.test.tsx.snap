// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should match the snapshot 1`] = `
<div>
  <div
    class="fixed top-0 right-0 bottom-0 left-0 flex items-center justify-center bg-tertiary z-50"
  >
    <span
      class="inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite]"
      role="status"
    />
  </div>
</div>
`;
